### PR TITLE
Harden cachedGitDiff: symlink prevention and file permissions (#94)

### LINKED ISSUE #94: Implement shared git context caching (ADR-018)
## Context
ADR-018 (shared git context for hook deduplication) was implemented in PR #91 with a temp file cache. However, the review flagged gaps:

- [RISK] Symlink TOCTOU on cache read (Szabo)
- [RISK] Timeout reduction from 5000ms to 2000ms untested on large repos (Knuth)
- Cache file is world-readable (0644) — should be 0600

## What's implemented
- cachedGitDiff() in tdd-enforce and pre-commit-gate hooks
- Temp file cache with 5s TTL, keyed by cwd hash + args
- 1 test for cache file creation

## What's missing
- [ ] Use fs.openSync with O_NOFOLLOW or lstatSync to prevent symlink attacks
- [ ] Write cache with mode 0o600
- [ ] Test: verify stale cache (>5s) triggers fresh git call
- [ ] Test: corrupted cache file falls through gracefully
- [ ] Document timeout trade-off in ADR consequences section
- [ ] Consider restoring 5000ms timeout for cold calls, 2000ms for cached reads

## Target: v0.6

### PR DESCRIPTION
**Pull request by fredericboyer**
## Summary
- Replaces `statSync` with `lstatSync` to detect and reject symlink cache files (prevents symlink attacks)
- Writes cache files with mode `0o600` (owner-only permissions)
- Updates ADR-018 with security hardening and timeout trade-off documentation
- Adds 4 new tests: stale cache, corrupted cache, symlink rejection, file permissions

## Details
Closes #94. Addresses security findings from the ADR-018 review:
- Szabo: symlink TOCTOU on cache read
- Knuth: missing test coverage for cache edge cases

Uses `lstatSync` instead of `O_NOFOLLOW` for better cross-platform portability (Node.js doesn't expose `O_NOFOLLOW` on all platforms).

## Test plan
- [x] Verify stale cache (>5s) triggers fresh git call
- [x] Verify corrupted cache file (directory at file path) falls through gracefully
- [x] Verify symlink cache files are rejected, removed, and target file preserved
- [x] Verify cache files written with mode 0o600
- [x] All 178 existing tests pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .claude/hooks/dev-team-pre-commit-gate.js [hunk 1]
```diff
@@ -30,16 +30,24 @@ function cachedGitDiff(args, timeoutMs) {
...
+    // file at a sensitive path and have us overwrite it on the next write)
+    if (stat.isSymbolicLink()) {
+      try {
+        fs.unlinkSync(cacheFile);
+      } catch {
+        /* best effort cleanup */
+      }
+    } else if (Date.now() - stat.mtimeMs < 5000) {
       return fs.readFileSync(cacheFile, "utf-8");
     }
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply .claude/hooks/dev-team-pre-commit-gate.js]: Fixed: implemented atomic temp-file + rename write strategy to close the TOCTOU window between lstat and write.

File: templates/hooks/dev-team-pre-commit-gate.js [hunk 2]
```diff
@@ -30,16 +30,24 @@ function cachedGitDiff(args, timeoutMs) {
...
+    const stat = fs.lstatSync(cacheFile);
+    // Reject symlinks to prevent symlink attacks (attacker could point cache
+    // file at a sensitive path and have us overwrite it on the next write)
+    if (stat.isSymbolicLink()) {
+      try {
+        fs.unlinkSync(cacheFile);
+      } catch {
+        /* best effort cleanup */
+      }
+    } else if (Date.now() - stat.mtimeMs < 5000) {
```
[the remark this answers is not in the document]
Comment (fredericboyer) [reply templates/hooks/dev-team-pre-commit-gate.js]: Fixed: unlink failure now sets skipWrite=true to disable cache write, and writes use atomic temp+rename.
